src/panels/lovelace/cards/hui-todo-list-card.ts

```diff
@@ -160,7 +160,10 @@ export class HuiTodoListCard extends LitElement implements LovelaceCard {
     (items?: TodoItem[], sort?: string | undefined): TodoItem[] =>
       items
         ? this._sortItems(
-            items.filter((item) => item.status === TodoItemStatus.Completed),
+            this._filterItemsByDueDate(
+              items.filter((item) => item.status === TodoItemStatus.Completed),
```

Comment:
We now filter twice, might be a good idea to combine the 2 filters and make a `getItems` function that accepts the `status` and `days_to_show` option.